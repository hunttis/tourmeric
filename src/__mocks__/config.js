export const config = {
  default: {
    titleText: 'Foo',
    apiKey: 'Foo',
    authDomain: 'Foo',
    databaseURL: 'Foo',
    projectId: 'Foo',
    storageBucket: 'Foo',
    messagingSenderId: 'Foo',
    fileMetadataFactory: (uploadRes) => {
      const { metadata: { name, fullPath, downloadURLs } } = uploadRes;
      return {
        name,
        fullPath,
        downloadURL: downloadURLs,
      };
    },
  },
};

exports.get = function get() {
  return config.default;
};
