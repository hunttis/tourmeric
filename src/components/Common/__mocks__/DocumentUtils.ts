
export default function scrollIntoview(_elementId: string) {
  console.log('scrolling');
}

export function setListener(_eventType: string, _callback: () => {}) {
  console.log('adding listener');
}

export function removeClassFromHtml(className: string) {
  console.log('removing ', className);
}

export function addClassToHtml(className: string) {
  console.log('adding ', className);
}