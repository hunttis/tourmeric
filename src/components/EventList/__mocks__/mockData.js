export const mockEventId = 'mockEventId1';

export const mockUserId = 'mockUserId';

export const mockProfile = {
  avatarUrl: 'https://foo.com/picture',
  dciNumber: '123512351112',
  displayName: 'Test Person',
  email: 'fake@fake.com',
  firstName: 'Test',
  lastName: 'Person',
  providerData: [{ displayName: 'Test Person',
    email: 'fake@fake.com',
    photoURL: 'https://foo.com/picture',
    providerId: 'facebook.com',
    uid: '1234567890' }],
  role: 'admin',
  isEmpty: false,
  isLoaded: true,
};

export const mockSingleEvent = {
  category: 'category1',
  createDate: '2018-05-24T17:00:46.500Z',
  date: '2018-09-24',
  entryFee: '10',
  format: 'Miniatures',
  link: 'http://www.google.fi',
  name: 'Warhammer thing',
  notes: 'Bring your war face!',
  playerSlots: '10',
  prizes: 'Glory!',
  published: true,
  rulesLevel: 'Casual',
  time: 'Kill everything',
};

export const mockUnorderedEvents = {
  mockEvent2: {
    category: 'category2',
    createDate: '2018-05-20T16:46:32.009Z',
    date: '2018-05-20',
    entryFee: '6',
    format: 'Standard',
    name: 'Tapahtuma',
    notes: 'FIJoijgasdsdgasdggagads',
    playerSlots: '14',
    prizes: 'Foooooooooooooooo',
    rulesLevel: 'Competitive',
    time: '10:00',
  },
  mockEvent3: {
    category: 'category2',
    createDate: '2018-05-20T16:04:29.595Z',
    date: '2018-05-21',
    entryFee: '5',
    format: 'Modern',
    link: 'https://magic.wizards.com/en/game-info/gameplay/formats/modern',
    name: 'Monday Modern',
    notes: 'Bring a friend!',
    playerSlots: '31',
    prizes: 'Three (3) OMG credits per player into the prizepool.',
    published: true,
    rulesLevel: 'Regular',
    time: '19:00',
  },
};

export const mockEvents = [
  { key: 'mockEvent2',
    value: {
      category: 'category2',
      createDate: '2018-05-20T16:46:32.009Z',
      date: '2018-05-20',
      entryFee: '6',
      format: 'Standard',
      name: 'Tapahtuma',
      notes: 'FIJoijgasdsdgasdggagads',
      playerSlots: '14',
      prizes: 'Foooooooooooooooo',
      rulesLevel: 'Competitive',
      time: '10:00',
    } },
  { key: 'mockEvent3',
    value: {
      category: 'category2',
      createDate: '2018-05-20T16:04:29.595Z',
      date: '2018-05-21',
      entryFee: '5',
      format: 'Modern',
      link: 'https://magic.wizards.com/en/game-info/gameplay/formats/modern',
      name: 'Monday Modern',
      notes: 'Bring a friend!',
      playerSlots: '31',
      prizes: 'Three (3) OMG credits per player into the prizepool.',
      published: true,
      rulesLevel: 'Regular',
      time: '19:00',
    } },
  { key: 'mockEvent4',
    value: {
      category: '-LDIw-1LFgiM0EJZGfus',
      createDate: '2018-05-24T20:58:06.938Z',
      date: '2018-05-24',
      entryFee: '5',
      format: 'League',
      link: 'jokulinkki.fi',
      name: 'Necromunda League',
      notes: 'Shame! Shame! Shame! *DING DING DING*',
      playerSlots: '12',
      prizes: 'I am Loot.',
      published: true,
      rulesLevel: 'Casual',
      time: '17:00',
    } },
  { key: 'mockEvent5',
    value: {
      category: '-LAlvCmIWNzYKpa7-7HI',
      createDate: '2018-05-24T17:01:53.909Z',
      date: '2018-05-26',
      entryFee: '0',
      format: 'Board game',
      link: 'https://geekandsundry.com/international-tabletop-day-april-28-2018/',
      name: 'Lautapeli Lauantai',
      notes: 'Bring a friend and get a surprise gift!',
      playerSlots: '30',
      prizes: "A Supi's surprise prize for all participants!",
      published: true,
      rulesLevel: 'Casual',
      time: '00:00',
    } },
  { key: 'mockEventId1',
    value: {
      category: 'category1',
      createDate: '2018-05-24T17:00:46.500Z',
      date: '2018-09-24',
      entryFee: '10',
      format: 'Miniatures',
      link: 'http://www.google.fi',
      name: 'Warhammer thing',
      notes: 'Bring your war face!',
      playerSlots: '10',
      prizes: 'Glory!',
      published: true,
      rulesLevel: 'Casual',
      time: 'Kill everything',
    } },
];

export const mockSettings = {
  activeLogo: 'http://foo.com/image.png',
  dateFormat: 'DD.MM.YYYY',
  pageSubtitle: '',
  pageTitle: '',
  subtitleTextColor: '#000',
  theme: 'darkly',
  titleBarAngle: '150',
  titleBarColor: '#000',
  titleBarColor2: '#ec1c22',
  titleBarPercentage: '50',
  titleTextColor: '',
};

export const mockUploadedLogos =
  {
    15271953483568546: { downloadURL: 'http://foo.com/image.png' },
    '152719372155055672': { downloadURL: 'http://foo.com/image.png' },
    '152719272536668394': { downloadURL: 'http://foo.com/image.png' },
    '1527227370219137123': { downloadURL: 'http://foo.com/image.png' },
    '152722965771313789': { downloadURL: 'http://foo.com/image.png' },
    '152722747904159438': { downloadURL: 'http://foo.com/image.png' },
  };

export const mockParticipations = { mockEventId1: { mockUserId: { comment: 'Jee',
  date: '2018-06-23T09:40:26.992Z',
  firstName: 'Test',
  lastName: 'Person',
  userId: 'mockUserId' } } };

export const mockCategories = { category1: { abbreviation: 'WH40K',
  logo: '1527227370219137123',
  name: 'Warhammer 40,000',
  type: 'Miniatures' },
category2: { abbreviation: 'MtG',
  logo: '152722965771313789',
  name: 'Magic: The Gathering',
  type: 'TCG' },
'-LAlvCmIWNzYKpa7-7HI': { abbreviation: 'Lautapelit',
  logo: '152719272536668394',
  name: 'Lautapelit / Board games' },
'-LCmESny8m9JVEFQ_qOt': { abbreviation: 'PKM',
  logo: '152722747904159438',
  name: 'Pok√©mon',
  type: 'TCG' },
'-LDImTAGxkzrorL0tQdi': { abbreviation: 'VG',
  logo: '152719372155055672',
  name: 'Videogames' },
'-LDIw-1LFgiM0EJZGfus': { abbreviation: 'Necromunda',
  logo: '15271953483568546',
  name: 'Necromunda',
  type: 'Miniatures' } };
