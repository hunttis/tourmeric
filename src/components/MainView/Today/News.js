
import React, { Component, Fragment } from 'react';
import { Translate } from 'react-localize-redux';
import { isLoaded } from 'react-redux-firebase';
import PropTypes from 'prop-types';
import moment from 'moment';
import _ from 'lodash';

export default class News extends Component {

  state = { activeNewsItem: '' }

  setActiveNewsItem(newsId) {
    this.setState({ activeNewsItem: newsId });
  }

  render() {
    const { news, settings } = this.props;

    const dateFormat = _.get(settings, 'dateFormat', 'DD-MM-YYYY');

    if (isLoaded(news)) {
      const publishedNews = news.filter(newsItem => newsItem.value.active);
      return (
        <Fragment>
          <h1 className="title"><Translate id="news" /></h1>
          <div>&nbsp;</div>
          {publishedNews.map((newsItem) => {
            const newsData = newsItem.value;
            const newsDate = moment(newsData.date, 'YYYY-MM-DD').format(dateFormat);
            const currentActiveNewsItem = newsItem.key === this.state.activeNewsItem;
            const formattedContent = newsData.text.split('\n');
            const longNewsItem = formattedContent.length > 3;
            const footerNeeded = newsData.link || longNewsItem;

            return (
              <Fragment key={newsItem.key}>
                <div className="card">
                  <div className="card-header">
                    <div className="card-header-title">
                      {newsData.name}
                    </div>
                    <div className="card-header-title">
                      {newsDate}
                    </div>
                  </div>

                  <div className="card-content">
                    <div className="content">
                      {(!longNewsItem || (longNewsItem && currentActiveNewsItem)) && formattedContent.map((paragraph, index) => <p key={`${newsItem.key}-${index}`}>{paragraph}</p>)}

                      {(longNewsItem && !currentActiveNewsItem) && formattedContent.map((paragraph, index) => {
                        if (index < 2 && !_.isEmpty(paragraph)) {
                          return <p key={`${newsItem.key}-${index}`}>{paragraph}&nbsp;</p>;
                        }
                        return '';
                      })}
                      {(longNewsItem && !currentActiveNewsItem) && <p>...<Translate id="continues" /></p>}
                    </div>
                  </div>

                  {footerNeeded &&
                    <div className="card-footer">
                      {(longNewsItem && !currentActiveNewsItem) &&
                        <a className="card-footer-item" onClick={() => this.setActiveNewsItem(newsItem.key)}><i className="fas fa-caret-square-down" />&nbsp;&nbsp;<Translate id="showfullnewsitem" /></a>
                      }
                      {currentActiveNewsItem &&
                        <a className="card-footer-item" onClick={() => this.setActiveNewsItem('')}><i className="fas fa-caret-square-up" />&nbsp;&nbsp;<Translate id="showless" /></a>
                      }
                      {(!newsData.summary && newsData.link) &&
                        <a className="card-footer-item" target="_blank" rel="noopener noreferrer" href={newsData.link}><i className="fas fa-external-link-alt" />&nbsp;&nbsp;{newsData.linkName ? newsData.linkName : newsData.link}</a>
                      }
                    </div>
                  }

                </div>
                <div>&nbsp;</div>
              </Fragment>
            );
          })}
        </Fragment>
      );
    }
    return <div><Translate id="loading" /></div>;
  }
}

News.propTypes = {
  news: PropTypes.array,
  settings: PropTypes.object,
};
