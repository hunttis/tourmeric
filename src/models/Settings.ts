
export interface Settings {
  activeLocationImage?: string;
  activeLogo: string;
  browserTitle?: string;
  companyinfo?: string;
  dateFormat: string;
  features?: Features;
  introText?: string;
  location?: Location;
  openingHours?: OpeningHours;
  pageSubtitle: string;
  pageTitle: string;
  privacyPolicy?: string;
  showSponsors?: boolean;
  subtitleTextColor: string;
  theme: string;
  titleBarAngle: string;
  titleBarColor: string;
  titleBarColor2: string;
  titleBarPercentage: string;
  titleTextColor: string;
  footer?: {[key: string]: Footer};
  locationImage?: string;
}

interface Features {
  events: Feature;
  highlights: Feature;
  storeinfo: Feature;
}

interface Feature {
  active: boolean;
  introtext: boolean;
}

interface Location {
  address: string;
  directions: string;
  email: string;
  phone: string;
}


export interface OpeningHours {
  additionalinfo: string;
  monday: string;
  tuesday: string;
  wednesday: string;
  thursday: string;
  friday: string;
  saturday: string;
  sunday: string;
}

export interface Footer {
  link: string;
  image: string;
  text: string;
}
