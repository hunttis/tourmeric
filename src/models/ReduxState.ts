import { Settings } from './Settings';
import { Category, UploadedFile } from './Category';
import { OpeningHoursException } from './OpeningHours';
import { TourmericEvent } from './Events';
import { TourmericStoreCreditData, StoreCreditCategory } from './StoreCredit';

export interface ReduxState {
  admin: any;
  firebase: FirebaseObject;
  locale: any;
  router: Router;
  editor: Editor;
}

interface Editor {
  returnLocation: string;
}

export interface FirebaseObject {
  data: FirebaseData;
  ordered: OrderedFirebaseData;
  auth: FirebaseAuth;
  profile: FirebaseProfile;
}

export interface Router {
  location: Location;
  action: string;
}

export interface Location {
  pathname: string;
  search: string;
  hash: string;
  key: string;
}

interface OrderedFirebaseData {
  events: [{ key: string, value: TourmericEvent }];
  eventsongoing: [{ key: string, value: TourmericEvent }];
  news: [{ key: string, value: SingleNewsItem }];
  users: [{ key: string, value: User }];
}

interface FirebaseData {
  settings: Settings;
  highlights: { [key: string]: HighLight };
  events: { [key: string]: TourmericEvent };
  eventsongoing: { [key: string]: TourmericEvent };
  categories: { [key: string]: Category };
  participations: { [key: string]: Participation };
  uploadedFiles: { [key: string]: UploadedFile };
  uploadedCategoryLogos: { [key: string]: UploadedFile };
  uploadedHighlightBanners: { [key: string]: UploadedFile };
  uploadedNewsImages: { [key: string]: UploadedFile };
  uploadedFooterItems: { [key: string]: UploadedFile };
  uploadedStoreinfoFiles: { [key: string]: UploadedFile };
  openinghoursexceptions: { [key: string]: OpeningHoursException };
  storecreditcategories: { [key: string]: StoreCreditCategory };
  users: { [key: string]: User };
  news: { [key: string]: SingleNewsItem };
  storecredit: { [key: string]: { [key: string]: TourmericStoreCreditData }};
}

export interface SingleNewsItem {
  createDate: string;
  date: string;
  linkName: string;
  link: string;
  image: string;
  active: boolean;
  name: string;
  text: string;
  summary: string;
}


export interface HighLight {
  active: boolean;
  image: string;
  createDate: string;
  date: string;
  name: string;
}

export interface Participation {
  [key: string]: ParticipationData;
}

export interface ParticipationData {
  date: string;
  firstName: string;
  lastName: string;
  userId: string;
  comment: string;
}

export interface FirebaseAuth {
  uid: string;
  displayName: string;
  photoURL: string;
  email: string | null;
  emailVerified: boolean;
  providerData: ProviderData;
  apiKey: string;
  isEmpty: boolean;
  isLoaded: boolean;
}

export interface ProviderData {
  displayName: string;
  email: string;
  photoURL: string;
  providerId: string;
  uid: string;
}

export interface FirebaseProfile {
  email: string;
  acceptedPrivacyPolicy?: string;
  acceptsNewsLetter?: boolean;
  avatarUrl: string;
  betaParticipation?: any;
  dciNumber: string;
  displayName: string;
  favoriteCategories?: string;
  firstName: string;
  lastName: string;
  landingPage?: PageOption;
  otherEmail?: string;
  providerData: ProviderData[];
  role: string;
  useOtherEmail?: boolean;
  isEmpty: boolean;
  isLoaded: boolean;
}

export type PageOption = 'today' | 'events' | 'storeinfo' | 'userinfo' | 'companyinfo' | 'admintools' | 'admintoolsevents' | 'adminsitesettings' | '';

export interface User {
  avatarUrl?: string;
  displayName?: string;
  providerData?: ProviderData;
  firstName: string;
  lastName: string;
  email: string;
  username: string;
  dciNumber?: string;
  role?: string;
  active: boolean;
}
