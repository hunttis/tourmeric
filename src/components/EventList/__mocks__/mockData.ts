import { FirebaseProfile, Participation, User, ProviderData, Location } from '~/models/ReduxState';
import { TourmericEvent } from '~/models/Events';
import { Settings } from '~/models/Settings';
import { UploadedFile, Category } from '~/models/Category';

export const mockEventId = 'mockEvent1';

export const mockUserId = 'mockUserId';

export const mockLocation: Location = {
  pathname: '/events/2018/10',
  search: 'search',
  hash: 'hash',
  key: 'key',
};

export const mockProviderData: ProviderData = {
  displayName: 'MockName',
  email: 'mock@userton.com',
  photoURL: 'http://url',
  providerId: 'facebook',
  uid: '1234567890',
};

export const mockUser: User = {
  avatarUrl: 'http://www.google.fi/google.png',
  displayName: 'Mock Userton',
  firstName: 'Mock',
  lastName: 'Userton',
  providerData: mockProviderData,
  email: 'mock@userton.com',
  username: 'Mockie',
  dciNumber: '123456',
  role: 'user',
  active: true,
};

export const mockProfile: FirebaseProfile = {
  avatarUrl: 'https://foo.com/picture',
  dciNumber: '123512351112',
  displayName: 'Test Person',
  email: 'fake@fake.com',
  firstName: 'Test',
  lastName: 'Person',
  providerData: [{
    displayName: 'Test Person',
    email: 'fake@fake.com',
    photoURL: 'https://foo.com/picture',
    providerId: 'facebook.com',
    uid: '1234567890',
  }],
  role: 'admin',
  isEmpty: false,
  isLoaded: true,
};

export const mockSingleEvent: TourmericEvent = {
  category: 'category1',
  createDate: '2018-05-24T17:00:46.500Z',
  date: '2018-09-24',
  entryFee: '10',
  format: 'Miniatures',
  link: 'http://www.google.fi',
  name: 'Warhammer thing',
  notes: 'Bring your war face!',
  playerSlots: '10',
  prizes: 'Glory!',
  published: true,
  rulesLevel: 'Casual',
  time: 'Kill everything',
  eventType: 'single',
};

export const mockUnorderedEvents: {[key: string]: TourmericEvent} = {
  mockEvent1: mockSingleEvent,
  mockEvent2: {
    category: 'category2',
    createDate: '2018-05-20T16:46:32.009Z',
    date: '2018-05-20',
    entryFee: '6',
    format: 'Standard',
    name: 'Tapahtuma',
    notes: 'FIJoijgasdsdgasdggagads',
    playerSlots: '14',
    prizes: 'Foooooooooooooooo',
    rulesLevel: 'Competitive',
    time: '10:00',
  },
  mockEvent3: {
    category: 'category2',
    createDate: '2018-05-20T16:04:29.595Z',
    date: '2018-05-21',
    entryFee: '5',
    format: 'Modern',
    link: 'https://magic.wizards.com/en/game-info/gameplay/formats/modern',
    name: 'Monday Modern',
    notes: 'Bring a friend!',
    playerSlots: '31',
    prizes: 'Three (3) OMG credits per player into the prizepool.',
    published: true,
    rulesLevel: 'Regular',
    time: '19:00',
  },
};

export const mockEvents: {key: string, value: TourmericEvent}[] = [
  { key: 'mockEvent1',
    value: mockSingleEvent },
  { key: 'mockEvent2',
    value: {
      category: 'category2',
      createDate: '2018-05-20T16:46:32.009Z',
      date: '2018-05-20',
      entryFee: '6',
      format: 'Standard',
      name: 'Tapahtuma',
      notes: 'FIJoijgasdsdgasdggagads',
      playerSlots: '14',
      prizes: 'Foooooooooooooooo',
      rulesLevel: 'Competitive',
      time: '10:00',
    } },
  { key: 'mockEvent3',
    value: {
      category: 'category2',
      createDate: '2018-05-20T16:04:29.595Z',
      date: '2018-05-21',
      entryFee: '5',
      format: 'Modern',
      link: 'https://magic.wizards.com/en/game-info/gameplay/formats/modern',
      name: 'Monday Modern',
      notes: 'Bring a friend!',
      playerSlots: '31',
      prizes: 'Three (3) OMG credits per player into the prizepool.',
      published: true,
      rulesLevel: 'Regular',
      time: '19:00',
    } },
  { key: 'mockEvent4',
    value: {
      category: '-LDIw-1LFgiM0EJZGfus',
      createDate: '2018-05-24T20:58:06.938Z',
      date: '2018-05-24',
      entryFee: '5',
      format: 'League',
      link: 'jokulinkki.fi',
      name: 'Necromunda League',
      notes: 'Shame! Shame! Shame! *DING DING DING*',
      playerSlots: '12',
      prizes: 'I am Loot.',
      published: true,
      rulesLevel: 'Casual',
      time: '17:00',
    } },
  { key: 'mockEvent5',
    value: {
      category: '-LAlvCmIWNzYKpa7-7HI',
      createDate: '2018-05-24T17:01:53.909Z',
      date: '2018-05-26',
      entryFee: '0',
      format: 'Board game',
      link: 'https://geekandsundry.com/international-tabletop-day-april-28-2018/',
      name: 'Lautapeli Lauantai',
      notes: 'Bring a friend and get a surprise gift!',
      playerSlots: '30',
      prizes: "A Supi's surprise prize for all participants!",
      published: true,
      rulesLevel: 'Casual',
      time: '00:00',
    } },
  { key: 'mockEventId1',
    value: {
      category: 'category1',
      createDate: '2018-05-24T17:00:46.500Z',
      date: '2018-09-24',
      entryFee: '10',
      format: 'Miniatures',
      link: 'http://www.google.fi',
      name: 'Warhammer thing',
      notes: 'Bring your war face!',
      playerSlots: '10',
      prizes: 'Glory!',
      published: true,
      rulesLevel: 'Casual',
      time: 'Kill everything',
    } },
];

export const mockSettings: Settings = {
  activeLogo: 'http://foo.com/image.png',
  dateFormat: 'DD.MM.YYYY',
  pageSubtitle: '',
  pageTitle: '',
  subtitleTextColor: '#000',
  theme: 'darkly',
  titleBarAngle: '150',
  titleBarColor: '#000',
  titleBarColor2: '#ec1c22',
  titleBarPercentage: '50',
  titleTextColor: '',
};

export const mockUploadedLogos: {[key: string]: UploadedFile} =
  {
    15271953483568546: { downloadURL: 'http://foo.com/image.png', name: 'image' },
    '152719372155055672': { downloadURL: 'http://foo.com/image.png', name: 'image' },
    '152719272536668394': { downloadURL: 'http://foo.com/image.png', name: 'image' },
    '1527227370219137123': { downloadURL: 'http://foo.com/image.png', name: 'image' },
    '152722965771313789': { downloadURL: 'http://foo.com/image.png', name: 'image' },
    '152722747904159438': { downloadURL: 'http://foo.com/image.png', name: 'image' },
  };

export const mockParticipations: {[key: string]: Participation} = {
  mockEventId1: {
    mockUserId: {
      comment: 'Jee',
      date: '2018-06-23T09:40:26.992Z',
      firstName: 'Test',
      lastName: 'Person',
      userId: 'mockUserId',
    },
  },
};

export const mockCategories: {[key: string]: Category} = {
  category1: {
    abbreviation: 'WH40K',
    logo: '1527227370219137123',
    name: 'Warhammer 40,000',
    type: 'Miniatures',
  },
  category2: {
    abbreviation: 'MtG',
    logo: '152722965771313789',
    name: 'Magic: The Gathering',
    type: 'TCG',
  },
  '-LAlvCmIWNzYKpa7-7HI': {
    abbreviation: 'Lautapelit',
    logo: '152719272536668394',
    name: 'Lautapelit / Board games',
    type: 'Boardgame',
  },
  '-LCmESny8m9JVEFQ_qOt': {
    abbreviation: 'PKM',
    logo: '152722747904159438',
    name: 'Pok√©mon',
    type: 'TCG',
  },
  '-LDImTAGxkzrorL0tQdi': {
    abbreviation: 'VG',
    logo: '152719372155055672',
    name: 'Videogames',
    type: 'Videogames',
  },
  '-LDIw-1LFgiM0EJZGfus': {
    abbreviation: 'Necromunda',
    logo: '15271953483568546',
    name: 'Necromunda',
    type: 'Miniatures',
  },
};
