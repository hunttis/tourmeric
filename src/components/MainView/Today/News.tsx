
import React from 'react';
import { Translate } from 'react-localize-redux';
import { isLoaded } from 'react-redux-firebase';
import _ from 'lodash';
import NewsItem from './NewsItem';
import { Settings } from '~/models/Settings';
import { SingleNewsItem } from '~/models/ReduxState';

interface Props {
  news: [{ key: string, value: SingleNewsItem }];
  settings: Settings;
}

export const News = ({ news, settings }: Props) => {

  const dateFormat = _.get(settings, 'dateFormat', 'DD-MM-YYYY');

  if (isLoaded(news)) {
    const publishedNews = news ? news.filter((newsItem) => newsItem.value.active) : [];
    return (
      <>
        <h1 className="title"><Translate id="news" /></h1>
        <div>&nbsp;</div>
        {publishedNews.map((newsItem) => <NewsItem key={`newsItem-${newsItem.key}`} newsItem={newsItem} dateFormat={dateFormat} />)}
      </>
    );
  }
  return <div><Translate id="loading" /></div>;

};
