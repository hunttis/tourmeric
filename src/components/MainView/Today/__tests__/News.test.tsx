/* eslint-disable import/first */

import React from 'react';
import { shallow } from 'enzyme';
import rrf from 'react-redux-firebase';
import { News } from '../News';
import { mockSettings } from '../../../../components/EventList/__mocks__/mockData';
import { Settings } from '../../../../models/Settings';
import { SingleNewsItem } from '../../../../models/ReduxState';

jest.mock('react-redux-firebase');

describe('News tests', () => {
  beforeEach(() => {
    rrf.isLoaded = jest.fn(() => true);
  });

  it('should render loading div', () => {

    rrf.isLoaded = jest.fn(() => false);

    const result = shallow(<News
      news={{} as any}
      settings={{} as Settings}
    />);

    expect(result).toMatchSnapshot();
  });

  it('should render empty news list', () => {
    rrf.isLoaded = jest.fn(() => true);

    const result = shallow(<News
      news={[] as any}
      settings={mockSettings}
    />);

    expect(result).toMatchSnapshot();
  });

  it('should render news list with two items', () => {
    rrf.isLoaded = jest.fn(() => true);

    const firstNewsItem: {key: string, value: SingleNewsItem} = {
      key: 'first',
      value: {
        createDate: '1-1-1990',
        date: '31-12-1990',
        linkName: 'Link',
        link: 'http://tosomesite.com',
        image: 'http://someimage/pic.png',
        active: true,
        name: 'First News Item',
        text: 'First News Item Text',
        summary: 'First Short summary',
      },
    };

    const secondNewsItem: {key: string, value: SingleNewsItem} = {
      key: 'second',
      value: {
        createDate: '1-1-2010',
        date: '31-12-2010',
        linkName: 'Other Link',
        link: 'http://tosomeothersite.com',
        image: 'http://someimage/pic2.png',
        active: true,
        name: 'Second News Item',
        text: 'Second News Item Text',
        summary: 'Second Short summary',
      },
    };

    const news: { key: string, value: SingleNewsItem }[] = [firstNewsItem, secondNewsItem];

    const result = shallow(<News
      news={news}
      settings={mockSettings}
    />);

    expect(result).toMatchSnapshot();
  });

});
