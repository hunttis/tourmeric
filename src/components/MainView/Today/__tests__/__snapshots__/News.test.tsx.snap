// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`News tests should render empty news list 1`] = `
<Fragment>
  <h1
    className="title"
  >
    <Translate
      id="news"
    />
  </h1>
  <div>
     
  </div>
</Fragment>
`;

exports[`News tests should render loading div 1`] = `
<div>
  <Translate
    id="loading"
  />
</div>
`;

exports[`News tests should render news list with two items 1`] = `
<Fragment>
  <h1
    className="title"
  >
    <Translate
      id="news"
    />
  </h1>
  <div>
     
  </div>
  <NewsItem
    dateFormat="DD.MM.YYYY"
    key="newsItem-first"
    newsItem={
      Object {
        "key": "first",
        "value": Object {
          "active": true,
          "createDate": "1-1-1990",
          "date": "31-12-1990",
          "image": "http://someimage/pic.png",
          "link": "http://tosomesite.com",
          "linkName": "Link",
          "name": "First News Item",
          "summary": "First Short summary",
          "text": "First News Item Text",
        },
      }
    }
  />
  <NewsItem
    dateFormat="DD.MM.YYYY"
    key="newsItem-second"
    newsItem={
      Object {
        "key": "second",
        "value": Object {
          "active": true,
          "createDate": "1-1-2010",
          "date": "31-12-2010",
          "image": "http://someimage/pic2.png",
          "link": "http://tosomeothersite.com",
          "linkName": "Other Link",
          "name": "Second News Item",
          "summary": "Second Short summary",
          "text": "Second News Item Text",
        },
      }
    }
  />
</Fragment>
`;
